sharding.go

```diff
@@ -266,8 +268,14 @@ func (s *Sharding) switchConn(db *gorm.DB) {
 	// When DoubleWrite is enabled, we need to query database schema
 	// information by table name during the migration.
 	if _, ok := db.Get(ShardingIgnoreStoreKey); !ok {
-		s.ConnPool = &ConnPool{ConnPool: db.Statement.ConnPool, sharding: s}
-		db.Statement.ConnPool = s.ConnPool
+		s.mutex.Lock()
+		if db.Statement.ConnPool != nil {
+			if _, ok = db.Statement.ConnPool.(ConnPool); !ok {
```

Comment:
Look like we don't need this line code? Just check `db.Statement.ConnPool != nil` is work.

Reply:
You're right. If `db.Statement.ConnPool != nil` was true, that means `db.Statement.ConnPool` has already set to `ConnPool` before.